File: testdata/jennies/builders/map_of_disjunctions/PHPBuilder/src/MapOfDisjunctions/DashboardBuilder.php
<?php

namespace Grafana\Foundation\MapOfDisjunctions;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\MapOfDisjunctions\Dashboard>
 */
class DashboardBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\MapOfDisjunctions\Dashboard $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\MapOfDisjunctions\Dashboard();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\MapOfDisjunctions\Dashboard
     */
    public function build()
    {
        return $this->internal;
    }

    /**
     * @param array<string, \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\MapOfDisjunctions\Element>> $panels
     */
    public function panels(array $panels): static
    {
            $panelsResources = [];
            foreach ($panels as $key1 => $val1) {
                    $panelsResources[$key1] = $val1->build();
            }
        $this->internal->panels = $panelsResources;
    
        return $this;
    }

}

File: testdata/jennies/builders/map_of_disjunctions/CSharpBuilder/src/Grafana/Foundation/MapOfDisjunctions/DashboardBuilder.cs
namespace Grafana.Foundation.MapOfDisjunctions;


public class DashboardBuilder : Cog.IBuilder<Dashboard>
{
    protected readonly Dashboard @internal;

    public DashboardBuilder()
    {
        this.@internal = new Dashboard();
    }

    public DashboardBuilder Panels(Dictionary<string, Cog.IBuilder<Element>> panels)
    {
        Dictionary<string, Element> panelsResources = new Dictionary<string, Element>();
        foreach (var entry1 in panels)
        {
                Element panelsDepth1 = entry1.Value.Build();
                panelsResources[entry1.Key] = panelsDepth1;
        }
        this.@internal.Panels = panelsResources;
        return this;
    }

    public Dashboard Build()
    {
        return this.@internal;
    }
}
